### ISSUE TITLE
multi-scale inference

### ISSUE THREAD

**Issue by qiulesun** [yz93/LAVT-RIS#7]
In the field of semantic segmentation, during training stage, more conventional augmentations are used, [like this](https://github.com/open-mmlab/mmsegmentation/blob/master/configs/_base_/datasets/ade20k.py#L7),  and during inference stage, [multi-scale + flipping results with scaling factors of [0.5; 0.75; 1.0; 1.25; 1.5; 1.75]](https://github.com/open-mmlab/mmsegmentation/blob/master/tools/test.py#L143) are commonly reported for obtaining SOTA. Have you considered adopting such strategies during training and inference to boost performence.

**Comment by yz93** [comment]
These are very good points! Thanks a lot for bringing them.

We did not try augmentations during training as we focused on the simplicity of the overall training pipeline. It probably would be reasonable to consider using augmentations like random graying, color jittering, and random cropping, although with a lot of care. Many expressions rely on color words to uniquely identify the object. Finding the optimal probability or range to vary the image colors can be a delicate matter, e.g., balancing between learning more robust features and risking losing discriminative information. Also, if applying random cropping, it would be important to ensure the target object is still present in the cropped image. We guess that cropping + ensuring that the target object makes up a large portion of the cropped image can be helpful as this would give us more positive pixels to train from in addition to augmented training data.

We did try multi-scale inference with scales [0.5, 1.0, 1.5]. This brings 0.1 to 0.5 absolute improvement in the overall IoU. We felt that these improvements are marginal, especially given the linearly growing inference time. So we did not mention this in the paper. But yes, this common inference strategy is helpful.

**Comment by qiulesun** [comment]
Many expressions rely on color words and  location words to uniquely identify the object. So the common augmentations (e.g., RandomCrop and RandomFlip) may not be useful, even degrading performance.  I agree with this view.

**Comment by yz93** [comment]
Hi, thanks!  --1. and 2. I see. Thanks! --3. Right. We uploaded a demo_inference.py file just now, which has some visualization code. Hope that works.

**Comment by qiulesun** [comment]
@yz93 The function [pad_if_smaller](https://github.com/yz93/LAVT-RIS/blob/main/transforms.py#L65) in transform script is not defined.

**Comment by yz93** [comment]
Thank you!

**Comment by qiulesun** [comment]
@yz93 Sorry for bothering you again. In Fig.6  of LAVT, you show the origianl input images, predicted masks and the ground-truth masks. It is straightforward for readers.  **Could you consider providing codes to obtaion such predicted masks and the ground-truth masks of **_entire datasets_** for analysis and observation ?**
